script.js

```diff
@@ -1,4 +1,275 @@
+var player1DiceRoll1 = 0;
+var player1DiceRoll2 = 0;
+var player2DiceRoll1 = 0;
+var player2DiceRoll2 = 0;
+var player1FinalDiceRollResult = 0;
+var player2FinalDiceRollResult = 0;
+var finalPlayer1AdvancedResults = 0;
+var finalPlayer2AdvancedResults = 0;
+var player1AdvanceRoll = [];
+var player2AdvanceRoll = [];
+var myOutputValue = "";
+var player1WinCount = 0;
+var player2WinCount = 0;
+var currentGameMode = "";
+
 var main = function (input) {
-  var myOutputValue = 'hello world';
+  //Output Gamemode choices "normal" and "advanced"
+  if (currentGameMode == "") {
+    myOutputValue =
+      "Please choose between the two gamemodes. Submit 'normal' for normal gamemode (i.e., 2 dices), and 'advanced' for advanced gamemode! (i.e., select no. of dices to be played, autochoice)";
+    currentGameMode = "Gamemode selected";
+    return myOutputValue;
+    //User selects Gamemode via gameMode function
+  } else if (currentGameMode == "Gamemode selected") {
+    var selectGameMode = gameMode(input);
+    myOutputValue = selectGameMode;
+    return myOutputValue;
+    //If gamemode is "Normal"
+  } else if (currentGameMode == "Submit") {
+    player1DiceRoll1 = rollDice();
+    player1DiceRoll2 = rollDice();
+    //play1 function shows result of dicerolls for player 1
+    var player1Roll = play1();
+    //chooseOrderMsg function shows choose order msg for player
+    var player1chooseOrderMsg = chooseOrderMsg();
+    myOutputValue = player1Roll + "<br>" + player1chooseOrderMsg;
+    currentGameMode = "waiting for player1 to choose order of dices";
+    return myOutputValue;
+  } else if (
+    currentGameMode == "waiting for player1 to choose order of dices"
+  ) {
+    //chooseOrderPlayer1 function outputs final result based on Player1's choice of the order of dices rolled
+    var player1ChooseOrder = chooseOrderPlayer1(input);
+    var myOutputValue =
+      player1ChooseOrder +
+      "<br> Player 2, please click on the submit button to begin!";
+    currentGameMode = "waiting for player2 to roll dices";
+    return myOutputValue;
+  } else if (currentGameMode == "waiting for player2 to roll dices") {
+    player2DiceRoll1 = rollDice();
+    player2DiceRoll2 = rollDice();
+    var player2Roll = play2();
```

Comment:
Same as line 33. You can consider changing the function name to more accurately describe what the function is doing. Remember that function names typically start with a verb: calculate, find, determine, choose etc...